ligo/main.mligo

```diff
@@ -105,27 +114,27 @@ let set_position (s : storage) (i_l : tick_index) (i_u : tick_index) (i_l_l : ti
     (* Get accumulated fees for this position. *)
     let s, fees = collect_fees s position_key in
     (* Update liquidity of position. *)
-    let liquidity_new = assert_nat (position.liquidity + delta_liquidity) in
+    let liquidity_new = assert_nat (position.liquidity + liquidity_delta) in
     let position = {position with liquidity = liquidity_new} in
     (* Reference counting the positions associated with a tick *)
     let ticks = (if liquidity_new = 0n then
         if is_new then
             ticks
         else
-            let ticks = incr_n_positions ticks i_l (-1) in
-            let ticks = incr_n_positions ticks i_u (-1) in
+            let ticks = cover_tick_with_position ticks i_l (-1) liquidity_delta in
+            let ticks = cover_tick_with_position ticks i_u (-1) (-liquidity_delta) in
             ticks
     else
         if is_new then
-            let ticks = incr_n_positions ticks i_l (1) in
-            let ticks = incr_n_positions ticks i_u (1) in
+            let ticks = cover_tick_with_position ticks i_l (1) (-liquidity_delta) in
+            let ticks = cover_tick_with_position ticks i_u (1) liquidity_delta in
```

Comment:
I think the signs are flipped in these 4 lines: `-liquidity_delta` should be `liquidity_delta`, and `liquidity_delta` should be `-liquidity_delta`

> It also adds liquidityDelta to the liquidityNet
value for the tick at the bottom end of the range, and subtracts it
from the liquidityNet at the upper tick

The way I'm reading this means that, in the second branch, where we're adding a new position, we should add `liquidity_delta` to `i_l`, and subtract `liquidity_delta` from `i_u`. In the first branch, it's the opposite.

Reply:
Oh right, the fact that "garbage collecting" branch goes first confused me :facepalm: